### PR TITLE
Update Tencent.WeChat to 3.3.5.42

### PR DESCRIPTION
**Pull request by VAllens**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/25109)

### CODE REVIEW

**Comment by VAllens** [comment]
I don't know what the problem is. 
The pipeline did not output any errors.
Can someone help me?

``` console
8/16/2021 1:57:52 AM

****************************************************************************************
Installation Verification
****************************************************************************************
Status: Started
============================================================================== 
Task         : Installation Validation
Description  : Invoke an Azure Function
Version      : 1.0.12
Author       : Microsoft Corporation
Help URL     : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/azure-function
============================================================================== 
2021-08-16T01:57:53.9217588Z POST ***?code=***&
Request body: {
  "operationId": "WinGetSvc-Validation-136-25109-20210816-1",
  "BuildId": "211950",
  "PlanUrl": "https://dev.azure.com/ms/",
  "HubName": "build",
  "ProjectId": "ed6a5dfa-6e7f-413b-842c-8305dd9e89e6",
  "PlanId": "cd8082c2-7185-4d37-a17a-01421ca6a7b4",
  "JobId": "871b4eed-22de-5cb6-c1c8-82f190de633f",
  "TimelineId": "cd8082c2-7185-4d37-a17a-01421ca6a7b4",
  "TaskInstanceId": "58bdf892-4e45-5ba7-4782-31912f95e167",
  "AuthToken": ***
}
				Response Code: OK
				Response: 

8/16/2021 1:58:25 AM

****************************************************************************************
Installation Verification
****************************************************************************************
Status: InProgress

8/16/2021 1:59:13 AM

****************************************************************************************
Installation Verification
****************************************************************************************
Status: InProgress

8/16/2021 2:00:17 AM

****************************************************************************************
Installation Verification
****************************************************************************************
Status: InProgress

8/16/2021 2:00:30 AM

****************************************************************************************
Installation Verification
****************************************************************************************
Status: Completed
Result: Failed
```

**Comment by lychichem** [comment]
Seems there is no version 3.3.5.42.

File: manifests/t/Tencent/WeChat/3.3.5.42/Tencent.WeChat.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+- silentWithProgress
+Installers:
+- InstallerLocale: en-US
+  Architecture: x86
+  InstallerType: nullsoft
+  Scope: machine
+  InstallerUrl: https://dldir1.qq.com/weixin/Windows/WeChatSetup.exe
+  InstallerSha256: 96D92B422535E4EDC88E0A6B6467FA783A5192068FC80ACF4845D4F988F8F09E
+  InstallerSwitches:
+    Custom: /NORESTART
```
Comment (lychichem) [inline manifests/t/Tencent/WeChat/3.3.5.42/Tencent.WeChat.installer.yaml:19]: the silent install switch is `/silent` . Tencent changed the default switch of Nullsoft installer, which cause the problem.

**Review by lychichem — COMMENTED** [review]
Please change the switch to `/silent /norestart` to see if it will solve the problem.

File: manifests/t/Tencent/WeChat/3.3.5.42/Tencent.WeChat.installer.yaml (hunk 1 above)
Comment (VAllens) [reply manifests/t/Tencent/WeChat/3.3.5.42/Tencent.WeChat.installer.yaml]: I'll give it a try, thanks. :)

**Comment by lychichem** [comment]
? On my machine, /silent works well

**Comment by VAllens** [comment]
> ? On my machine, /silent works well
> @VAllens you may need to change SILENT to silent, the switch is case sensitive.

Let me try again.
Thanks. @lychichem

**Review by OfficialEsco — APPROVED** [review]
Weird how /S failed?
image

**Comment by lychichem** [comment]
> Weird how /S failed?
> image

On my machine, /S does nothing, /Silent will install the package.
